packages/expo-notifications/ios/EXNotifications/Permissions/PermissionsModule.swift

```diff
@@ -0,0 +1,81 @@
+//  Copyright © 2024 650 Industries. All rights reserved.
+
+import ExpoModulesCore
+import UIKit
+import MachO
+
+public class PermissionsModule: Module {
+  var permissionsManager: (any EXPermissionsInterface)?
+  var requester: EXUserFacingNotificationsPermissionsRequester =
+    EXUserFacingNotificationsPermissionsRequester()
+
+  public func definition() -> ModuleDefinition {
+    Name("ExpoNotificationPermissionsModule")
+
+    OnCreate {
+      appContext?.permissions?.register([
+        requester
+      ])
+    }
+
+    AsyncFunction("getPermissionsAsync") { (promise: Promise) in
+      appContext?
+        .permissions?
+        .getPermissionUsingRequesterClass(
+          EXUserFacingNotificationsPermissionsRequester.self,
+          resolve: promise.resolver,
+          reject: promise.legacyRejecter
+        )
+    }
+
+    AsyncFunction("requestPermissionsAsync") { (requestedPermissions: NotificationPermissionRecord, promise: Promise) in
+      EXUserFacingNotificationsPermissionsRequester.setRequestedPermissions(requestedPermissions.toDictionary())
+      appContext?
+        .permissions?
+        .askForPermission(
+          usingRequesterClass: EXUserFacingNotificationsPermissionsRequester.self,
+          resolve: promise.resolver,
+          reject: promise.legacyRejecter
+        )
+    }
+  }
+}
+
+struct NotificationPermissionRecord: Record {
+  @Field
+  var allowAlert: Bool?
+  @Field
+  var allowBadge: Bool?
+  @Field
+  var allowSound: Bool?
+  @Field
+  var allowDisplayInCarPlay: Bool?
+  @Field
+  var allowCriticalAlerts: Bool?
+  @Field
+  var provideAppNotificationSettings: Bool?
+  @Field
+  var allowProvisional: Bool?
+
+  func count() -> Int {
+    return (self.allowAlert != nil ? 1 : 0) +
```

Comment:
this doesn't seem to be used

If it's needed somewhere (where?), maybe `self.toDictionary().count` would work the same

Reply:
I did some refactoring to simplify and to use this method, allowing me to remove code from the old Objective C file.